include/motor-control/firmware/brushed_motor/brushed_motor_hardware.hpp

```diff
@@ -18,20 +20,29 @@ struct BrushedHardwareConfig {
     gpio::PinConfig enable;
     gpio::PinConfig limit_switch;
     gpio::PinConfig sync_in;
+    double encoder_interrupt_freq;
+    double pid_kp;
+    double pid_ki;
+    double pid_kd;
 };
 
+enum class ControlDirection { positive, negative, unset };
+
 class BrushedMotorHardware : public BrushedMotorHardwareIface {
   public:
     ~BrushedMotorHardware() final = default;
     BrushedMotorHardware() = delete;
     BrushedMotorHardware(const BrushedHardwareConfig& config,
                          void* encoder_handle)
-        : pins(config), enc_handle(encoder_handle) {}
-    BrushedMotorHardware(const BrushedMotorHardware&) = default;
+        : pins(config),
+          enc_handle(encoder_handle),
+          controller_loop{config.pid_kp, config.pid_ki, config.pid_kd,
+                          1 / config.encoder_interrupt_freq} {}
```

Comment:
```suggestion
                          1.f / config.encoder_interrupt_freq} {}
```

just to make it explicit